File: leetcode/C++/summary-ranges.cpp
class Solution {
public:
    string makeRange(int &a, int &b){
        if(a==b) return to_string(a);
        return to_string(a)+"->"+to_string(b);
    }
    vector<string> summaryRanges(vector<int>& nums) {
        vector<string> sol;
        if(nums.empty()) return sol;
        int st = nums[0];
        for(int i=1; i<nums.size(); ++i){
            if(nums[i]!=nums[i-1]+1){
                sol.push_back(makeRange(st, nums[i-1]));
                st = nums[i];
            }
        }
        sol.push_back(makeRange(st, nums.back()));
        return sol;
    }
};

File: leetcode/Python/summary-ranges.py
class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        def subRange(a, b):
            return str(a)+'->'+str(b) if a!=b else str(a)
        sol, prev = list(), 0
        if nums is None or len(nums) == 0:
            return sol
        prev = nums[0]
        for i in range(1, len(nums)):
            if nums[i]!=nums[i-1]+1:
                sol.append(subRange(prev, nums[i-1]))
                prev = nums[i]
        sol.append(subRange(prev, nums[-1]))
        return sol